dispatch/dispatch.go

```diff
@@ -39,24 +35,43 @@ func init() {
 	prometheus.MustRegister(requestMtc)
 }
 
+type message interface {
+	ChainID() uint32
+}
+
 // blockMsg packages a proto block message.
 type blockMsg struct {
+	chainID uint32
```

Comment:
I thought our agreement is not to redefine the chain ID in the app level msg, because it's already available at network msg. Same for the following msgs.